Ddf 1244

brendan-hofmann opened the pull request:
Admin UI Metrics tab in the Platform app.

Note that this also includes adding UI access to the functionality implemented in DDF-1243.

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/codice/ddf-platform/62)
<!-- Reviewable:end -->

brendan-hofmann commented:
@pklinef @djblue @clockard

clockard commented:
The new /metrics context path needs to be added to the security-policy-context.

brendan-hofmann commented:
Good catch, I'll have to make a pull request for that.

clockard commented:
When selecting a PNG for one of the metrics it opens up within the iframe and gets cut off with no way to scroll to see the rest of it. Seems like this should either scale to fit within the viewable window, have scroll bars to so the whole image can be viewed or just go straight to download and not show the image. The current behavior is consistent with the original metrics in that it just opens up the png but it might be better if we scaled it to fit and include a "back" type button to get back to the metrics without the need to use the browser back button.

brendan-hofmann commented:
What do you think of a modal popup? That might be more consistent with the rest of our UI.

clockard commented:
Sounds good to me.

clockard commented:
Reviewed 22 of 22 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks broke.

---

<sup>**[metrics/platform-metrics-ui/pom.xml, line 42 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/62#-JsRIf4KhC2vd47LPmjL)** ([raw file](https://github.com/codice/ddf-platform/blob/97c7957eae9297e90260f0bc56b6701f8980801c/metrics/platform-metrics-ui/pom.xml#L42)):</sup>
Should the admin version dependencies be specified in the platform properties so we don't forget about updating the snapshots for releases? We could also do what the security admin module does and just use an older release version since the api doesn't change.

---

<sup>**[metrics/platform-metrics-ui/src/main/webapp/js/model/Metrics.js, line 31 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/62#-JsRM1IBZJkB1Tg6qeGN)** ([raw file](https://github.com/codice/ddf-platform/blob/97c7957eae9297e90260f0bc56b6701f8980801c/metrics/platform-metrics-ui/src/main/webapp/js/model/Metrics.js#L31)):</sup>
In some cases I could see a '12' count option as useful.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/62)
<!-- Sent from Reviewable.io -->

clockard commented:
Downloaded and built built branch. All tests ran all successfully.

brendan-hofmann commented:
Review status: 22 of 23 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

<sup>**[metrics/platform-metrics-ui/pom.xml, line 42 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/62#-JsRIf4KhC2vd47LPmjL)** ([raw file](https://github.com/codice/ddf-platform/blob/97c7957eae9297e90260f0bc56b6701f8980801c/metrics/platform-metrics-ui/pom.xml#L42)):</sup>
Both good ideas, I'll fix it.

---

<sup>**[metrics/platform-metrics-ui/src/main/webapp/js/model/Metrics.js, line 31 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/62#-JsRM1IBZJkB1Tg6qeGN)** ([raw file](https://github.com/codice/ddf-platform/blob/97c7957eae9297e90260f0bc56b6701f8980801c/metrics/platform-metrics-ui/src/main/webapp/js/model/Metrics.js#L31)):</sup>
Higher than six and you start having redundant overlaps (12 months = 1 year), which I'm not really a fan of.

The provided ranges are based on user requests. Arbitrary ranges and counts can still be accessed easily with a bit of math, using the custom URLs.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/62)
<!-- Sent from Reviewable.io -->

clockard commented:
Should the platform-metrics-webcosole-plugin be removed now that we have the admin ui metrics plugin?

brendan-hofmann commented:
Not yet. We're leaving the legacy interface alone for now, in case someone is really attached to using it. It'll go away when we get rid of the felix web console entirely.

brendan-hofmann commented on metrics/platform-metrics-reporting/README.md:
```diff
@@ -0,0 +1,6 @@
+## Sample Data
```
You can use this to generate some data so that you have something to look at in the UI.

brendan-hofmann commented on metrics/platform-metrics-ui/src/main/webapp/js/view/Metrics.view.js:
```diff
@@ -0,0 +1,111 @@
+                endDate.setUTCSeconds(0);
+                endDate.setUTCMilliseconds(0);
+                var startDate = new Date(endDate);
+                while (range > 0) {
```
After talking with @kguerriero, this math doesn't look right to me. I'll fix it.

pklinef commented:
Review status: 20 of 28 files reviewed at latest revision, 11 unresolved discussions, some commit checks broke.

---

<sup>**[metrics/platform-metrics-ui/pom.xml, line 47 \[r4\]](https://reviewable.io:443/reviews/codice/ddf-platform/62#-JsW4T17TWGaThh9bizC)** ([raw file](https://github.com/codice/ddf-platform/blob/85896dec0beb7f4054d3e33675ce3ad9d436a4bc/metrics/platform-metrics-ui/pom.xml#L47)):</sup>
I think this creates a circular dependency since admin ui depends on platform and is a separate repo.  This will cause a problem if we need to use the latest version.

---

<sup>**[metrics/platform-metrics-ui/src/main/java/org/codice/ddf/platform/ui/MetricsPlugin.java, line 30 \[r4\]](https://reviewable.io:443/reviews/codice/ddf-platform/62#-JsWCPlVUlnQepcgcQnd)** ([raw file](https://github.com/codice/ddf-platform/blob/85896dec0beb7f4054d3e33675ce3ad9d436a4bc/metrics/platform-metrics-ui/src/main/java/org/codice/ddf/platform/ui/MetricsPlugin.java#L30)):</sup>
This is fine but you might consider Arrays.asList().

http://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html#asList(T...)

---

<sup>**[metrics/platform-metrics-ui/src/main/webapp/js/view/Metrics.view.js, line 46 \[r4\]](https://reviewable.io:443/reviews/codice/ddf-platform/62#-JsWPpoweQ5SJX0GKSCI)** ([raw file](https://github.com/codice/ddf-platform/blob/85896dec0beb7f4054d3e33675ce3ad9d436a4bc/metrics/platform-metrics-ui/src/main/webapp/js/view/Metrics.view.js#L46)):</sup>
Instead of using jQuery, this information should be in the model.  You can use a model binder to hook input elements into the model.

---

<sup>**[metrics/platform-metrics-ui/src/main/webapp/js/view/Metrics.view.js, line 50 \[r4\]](https://reviewable.io:443/reviews/codice/ddf-platform/62#-JsWQ8IGw7JdjxgzOG3Z)** ([raw file](https://github.com/codice/ddf-platform/blob/85896dec0beb7f4054d3e33675ce3ad9d436a4bc/metrics/platform-metrics-ui/src/main/webapp/js/view/Metrics.view.js#L50)):</sup>
Most of this logic looks like it is related to the model and not the view.  I would suggest putting it into the model instead.

---

<sup>**[metrics/platform-metrics-ui/src/main/webapp/js/view/Metrics.view.js, line 98 \[r4\]](https://reviewable.io:443/reviews/codice/ddf-platform/62#-JsWQL65814BUE_TDhvw)** ([raw file](https://github.com/codice/ddf-platform/blob/85896dec0beb7f4054d3e33675ce3ad9d436a4bc/metrics/platform-metrics-ui/src/main/webapp/js/view/Metrics.view.js#L98)):</sup>
Why was this left here?

---

<sup>**[metrics/platform-metrics-ui/src/main/webapp/js/view/Metrics.view.js, line 100 \[r4\]](https://reviewable.io:443/reviews/codice/ddf-platform/62#-JsWQPxAZFo8WBlmJZZs)** ([raw file](https://github.com/codice/ddf-platform/blob/85896dec0beb7f4054d3e33675ce3ad9d436a4bc/metrics/platform-metrics-ui/src/main/webapp/js/view/Metrics.view.js#L100)):</sup>
This will blow up on some versions of IE.  You need to check if console and log is available before using it.

---

<sup>**[metrics/platform-metrics-ui/src/main/webapp/js/view/Metrics.view.js, line 104 \[r4\]](https://reviewable.io:443/reviews/codice/ddf-platform/62#-JsWQfGqIfv0YS8u56MC)** ([raw file](https://github.com/codice/ddf-platform/blob/85896dec0beb7f4054d3e33675ce3ad9d436a4bc/metrics/platform-metrics-ui/src/main/webapp/js/view/Metrics.view.js#L104)):</sup>
Instead of changing the location when clicking the link, you can change the href of the link based on the model as different values are selected.

---

<sup>**[metrics/platform-metrics-ui/src/main/webapp/less/styles.less, line 67 \[r4\]](https://reviewable.io:443/reviews/codice/ddf-platform/62#-JsWUJbBNpDDkXnyq48E)** ([raw file](https://github.com/codice/ddf-platform/blob/85896dec0beb7f4054d3e33675ce3ad9d436a4bc/metrics/platform-metrics-ui/src/main/webapp/less/styles.less#L67)):</sup>
Why did you not use hyphens for all style names?

---

<sup>**[security/policy/security-policy-context/src/main/resources/OSGI-INF/blueprint/blueprint.xml, line 25 \[r4\]](https://reviewable.io:443/reviews/codice/ddf-platform/62#-JsWVoZh1ESmMm6IKjOr)** ([raw file](https://github.com/codice/ddf-platform/blob/85896dec0beb7f4054d3e33675ce3ad9d436a4bc/security/policy/security-policy-context/src/main/resources/OSGI-INF/blueprint/blueprint.xml#L25)):</sup>
We should consider putting all admin modules under a common context.  Then we can configure the common context instead of having to add every new module to the policy.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/62)
<!-- Sent from Reviewable.io -->

brendan-hofmann commented:
Review status: 20 of 28 files reviewed at latest revision, 10 unresolved discussions, some commit checks broke.

---

<sup>**[metrics/platform-metrics-ui/pom.xml, line 47 \[r4\]](https://reviewable.io:443/reviews/codice/ddf-platform/62#-JsW4T17TWGaThh9bizC)** ([raw file](https://github.com/codice/ddf-platform/blob/85896dec0beb7f4054d3e33675ce3ad9d436a4bc/metrics/platform-metrics-ui/pom.xml#L47)):</sup>
We can depend on the release  version for now, but we really need to combine these repos sooner rather than later.

---

<sup>**[metrics/platform-metrics-ui/src/main/java/org/codice/ddf/platform/ui/MetricsPlugin.java, line 30 \[r4\]](https://reviewable.io:443/reviews/codice/ddf-platform/62#-JsWCPlVUlnQepcgcQnd)** ([raw file](https://github.com/codice/ddf-platform/blob/85896dec0beb7f4054d3e33675ce3ad9d436a4bc/metrics/platform-metrics-ui/src/main/java/org/codice/ddf/platform/ui/MetricsPlugin.java#L30)):</sup>
Good point.

---

<sup>**[metrics/platform-metrics-ui/src/main/webapp/js/view/Metrics.view.js, line 46 \[r4\]](https://reviewable.io:443/reviews/codice/ddf-platform/62#-JsWPpoweQ5SJX0GKSCI)** ([raw file](https://github.com/codice/ddf-platform/blob/85896dec0beb7f4054d3e33675ce3ad9d436a4bc/metrics/platform-metrics-ui/src/main/webapp/js/view/Metrics.view.js#L46)):</sup>
Good idea.

---

<sup>**[metrics/platform-metrics-ui/src/main/webapp/js/view/Metrics.view.js, line 50 \[r4\]](https://reviewable.io:443/reviews/codice/ddf-platform/62#-JsWQ8IGw7JdjxgzOG3Z)** ([raw file](https://github.com/codice/ddf-platform/blob/85896dec0beb7f4054d3e33675ce3ad9d436a4bc/metrics/platform-metrics-ui/src/main/webapp/js/view/Metrics.view.js#L50)):</sup>
Good idea.

---

<sup>**[metrics/platform-metrics-ui/src/main/webapp/js/view/Metrics.view.js, line 98 \[r4\]](https://reviewable.io:443/reviews/codice/ddf-platform/62#-JsWQL65814BUE_TDhvw)** ([raw file](https://github.com/codice/ddf-platform/blob/85896dec0beb7f4054d3e33675ce3ad9d436a4bc/metrics/platform-metrics-ui/src/main/webapp/js/view/Metrics.view.js#L98)):</sup>
Brain fart. ;)

---

<sup>**[metrics/platform-metrics-ui/src/main/webapp/js/view/Metrics.view.js, line 100 \[r4\]](https://reviewable.io:443/reviews/codice/ddf-platform/62#-JsWQPxAZFo8WBlmJZZs)** ([raw file](https://github.com/codice/ddf-platform/blob/85896dec0beb7f4054d3e33675ce3ad9d436a4bc/metrics/platform-metrics-ui/src/main/webapp/js/view/Metrics.view.js#L100)):</sup>
Removed.

---

<sup>**[metrics/platform-metrics-ui/src/main/webapp/js/view/Metrics.view.js, line 104 \[r4\]](https://reviewable.io:443/reviews/codice/ddf-platform/62#-JsWQfGqIfv0YS8u56MC)** ([raw file](https://github.com/codice/ddf-platform/blob/85896dec0beb7f4054d3e33675ce3ad9d436a4bc/metrics/platform-metrics-ui/src/main/webapp/js/view/Metrics.view.js#L104)):</sup>
Good idea.

---

<sup>**[metrics/platform-metrics-ui/src/main/webapp/less/styles.less, line 67 \[r4\]](https://reviewable.io:443/reviews/codice/ddf-platform/62#-JsWUJbBNpDDkXnyq48E)** ([raw file](https://github.com/codice/ddf-platform/blob/85896dec0beb7f4054d3e33675ce3ad9d436a4bc/metrics/platform-metrics-ui/src/main/webapp/less/styles.less#L67)):</sup>
Because I copied and modified another style and didn't proofread it  thoroughly enough. I don't even use that style. :(

---

<sup>**[security/policy/security-policy-context/src/main/resources/OSGI-INF/blueprint/blueprint.xml, line 25 \[r4\]](https://reviewable.io:443/reviews/codice/ddf-platform/62#-JsWVoZh1ESmMm6IKjOr)** ([raw file](https://github.com/codice/ddf-platform/blob/85896dec0beb7f4054d3e33675ce3ad9d436a4bc/security/policy/security-policy-context/src/main/resources/OSGI-INF/blueprint/blueprint.xml#L25)):</sup>
I'll try /admin/metrics.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/62)
<!-- Sent from Reviewable.io -->

clockard commented:
Reviewed 1 of 1 files at r2, 5 of 5 files at r3, 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 9 unresolved discussions, some commit checks broke.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/62)
<!-- Sent from Reviewable.io -->

pklinef commented:
:+1:

stustison commented on security/handler/security-handler-anonymous/bower.json:
```diff
@@ -22,7 +22,6 @@
   "resolutions": {
-    "jquery": "~1.11.0",
-    "underscore": "~1.8.2"
+    "jquery": "~1.11.0"
```
Remove this commit please. This needs to be reverted.

brendan-hofmann replied:
Successfully avoided for every commit except the last one. :(
